solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/JCSMPInboundChannelAdapter.java

```diff
@@ -107,6 +108,18 @@ protected void doStart() {
             throw exception;
         }
 
+        Map<String, String> headerNameMapping = consumerProperties.getExtension().getHeaderNameMapping();
+        if (headerNameMapping != null && !headerNameMapping.isEmpty()) {
+            Set<String> targetHeaderNames = new HashSet<>(headerNameMapping.values());
+            if (targetHeaderNames.size() < headerNameMapping.size()) {
+                MessagingException exception = new MessagingException(String.format(
```

Comment:
I know we threw other `start` exceptions as `MessagingException`, but in hindsight, this should be an `IllegalArgumentException` I think